File: app/src/main/java/de/tum/in/tumcampus/models/DeviceRegister.java
package de.tum.in.tumcampus.models;

import android.content.Context;

import java.math.BigInteger;
import java.security.SecureRandom;
import java.util.Date;

import de.tum.in.tumcampus.auxiliary.AuthenticationManager;
import de.tum.in.tumcampus.exceptions.NoPrivateKey;

public class DeviceRegister {

    private String signature;
    private String date;
    private String rand;
    private String device;
    private String publicKey;

    public DeviceRegister(Context c, String publickey) throws NoPrivateKey {
        //Create some data
        this.date = (new Date()).toString();
        this.rand = new BigInteger(130, new SecureRandom()).toString(32);
        this.device = AuthenticationManager.getDeviceID(c);
        this.publicKey = publickey;

        //Sign this data for verification
        AuthenticationManager am = new AuthenticationManager(c);
        this.signature = am.sign(date + rand + this.device);
    }

}

File: app/src/main/java/de/tum/in/tumcampusapp/api/app/model/DeviceRegister.kt
package de.tum.`in`.tumcampusapp.api.app.model

import android.content.Context
import de.tum.`in`.tumcampusapp.api.app.AuthenticationManager
import de.tum.`in`.tumcampusapp.api.app.exception.NoPrivateKey
import java.math.BigInteger
import java.security.SecureRandom
import java.util.*

data class DeviceRegister(
    var signature: String = "",
    var date: String = "",
    var rand: String = "",
    var device: String = "",
    var publicKey: String = ""
) {
    companion object {
        @Throws(NoPrivateKey::class)
        fun getDeviceRegister(c: Context, publickey: String): DeviceRegister {
            val am = AuthenticationManager(c)
            val date = Date().toString()
            val rand = BigInteger(130, SecureRandom()).toString(32)
            val device = AuthenticationManager.getDeviceID(c)

            return DeviceRegister(date = date, rand = rand, device = device, publicKey = publickey, signature = am.sign(date + rand + device))
        }
    }
}
